pkg/data/domain/app/service.go

```diff
@@ -0,0 +1,55 @@
+package app
+
+import (
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/app-operator/v2/service/controller/app/values"
+
+	"github.com/giantswarm/kubectl-gs/pkg/data/client"
+)
+
+var _ Interface = &Service{}
+
+type Config struct {
+	Client *client.Client
+	Logger micrologger.Logger
+}
+
+type Service struct {
+	client        *client.Client
+	valuesService *values.Values
+
+	catalogFetchResults map[string]CatalogFetchResult
+	schemaFetchResults  map[string]SchemaFetchResult
```

Comment:
The data service should only be used to fetch resources from the k8s API (hence the package name `data`). It shouldn't include additional business logic, such as caching results (here), or the app validation logic inside of `pkg/data/domain/app/validate.go`.

This business logic should live inside the command that uses it, or even better, inside a separate package in the `pkg` folder.

Reply:
Oh makes sense!

I moved this code now to `pkg/app` . There are a few places where it calls out to the k8s API, but it's pretty straightforward. Not sure it makes sense to extract that then into a specific data package just yet.